File: backend/src/main/java/com/example/backend/models/ReservationsView.java
package com.example.backend.models;

import java.util.List;

public class ReservationsView {
    List<Reservation> activeReservations;
    List<Reservation> finishedReservations;

    public ReservationsView() {
    }

    public ReservationsView(List<Reservation> activeReservations, List<Reservation> finishedReservations) {
        this.activeReservations = activeReservations;
        this.finishedReservations = finishedReservations;
    }

    public List<Reservation> getActiveReservations() {
        return activeReservations;
    }

    public void setActiveReservations(List<Reservation> activeReservations) {
        this.activeReservations = activeReservations;
    }

    public List<Reservation> getFinishedReservations() {
        return finishedReservations;
    }

    public void setFinishedReservations(List<Reservation> finishedReservations) {
        this.finishedReservations = finishedReservations;
    }
}

File: frontend/src/app/models/ReservationsView.ts
// src/app/models/reservations-view.model.ts

import { Reservation } from './Reservation';

export class ReservationsView {
  activeReservations: Reservation[];
  finishedReservations: Reservation[];

  constructor(
    activeReservations: Reservation[],
    finishedReservations: Reservation[]
  ) {
    this.activeReservations = activeReservations;
    this.finishedReservations = finishedReservations;
  }
}
